app/models/concerns/taxonomix.rb

```diff
@@ -132,6 +132,59 @@ def scope_by_taxable_ids(scope)
         scope.where("#{table_name}.id IN (#{cached_ids.join(',')})")
       end
     end
+
+    def join_organizations
+      join_taxonomies(user_organizations, "tto")
+    end
+
+    def join_locations
+      join_taxonomies(user_locations, "ttl")
+    end
+
+    def join_taxonomies(taxonomies, join_name)
+      joins("LEFT JOIN #{TAXONOMY_JOIN_TABLE} AS #{join_name} "\
```

Comment:
can we do this through arel as well? I'd rather not mix arel and pure sql if possible and this joins should not be as hard, right? We could pass in the arel table (`org_join_arel` or `loc_join_arel`)
I think if we do it through arel, the Rails have it easier to figure wheter the join is already used in a scope or not.

Reply:
yup. can do. shouldn't be an issue

Reply:
Done!